torch/quantization/_equalize.py

```diff
@@ -0,0 +1,133 @@
+
+    output_channel_tensor1 = channel_range(tensor1, output_axis)
+    input_channel_tensor2 = channel_range(tensor2, input_axis)
+
+    # producing scaling factors to applied
+    input_channel_tensor2 += 1e-9
+    scaling_factors = torch.sqrt(output_channel_tensor1 / input_channel_tensor2)
+    r_scaling_factors = torch.reciprocal(scaling_factors)
+
+    bias = bias * r_scaling_factors
+
+    # formatting the scaling (1D) tensors to be applied on the given argument tensors
+    # pads axis to (1D) tensors to then be broadcasted
+    size1 = [1 for x in range(len(tensor1.size()))]
+    size1[output_axis] = tensor1.size()[output_axis]
+    size2 = [1 for x in range(len(tensor2.size()))]
+    size2[input_axis] = tensor2.size()[input_axis]
+
+    scaling_factors = torch.reshape(scaling_factors, size2)
+    r_scaling_factors = torch.reshape(r_scaling_factors, size1)
+
+    tensor1 = tensor1 * r_scaling_factors
+    tensor2 = tensor2 * scaling_factors
+
+    module1.weight.data = tensor1
+    module1.bias.data = bias
+    module2.weight.data = tensor2
+
+def cross_layer_equalization(module1, module2):
+    ''' Given two adjacent modules, the weights are scaled such that
+    the ranges of the first modules' output channel are equal to the
+    ranges of the second modules' input channel
+
+    Scaling work is done in scaling_channels()
+    '''
+    module1_output_channel_axis = -1
+    module2_input_channel_axis = -1
+
+    # additional modules can be added, given the axises of the input/output channels are known
+    if isinstance(module1, nn.Linear) or isinstance(module1, nn.Conv2d):
+        module1_output_channel_axis = 0
+    else:
+        raise TypeError("Only Linear and Conv2d modules are supported at this time")
+    if isinstance(module2, nn.Linear) or isinstance(module2, nn.Conv2d):
+        module2_input_channel_axis = 1
+    else:
+        raise TypeError("Only Linear and Conv2d modules are supported at this time")
+
+    if module1.weight.size()[module1_output_channel_axis] != module2.weight.size()[module2_input_channel_axis]:
+        raise TypeError("Incompatible tensors")
+
+    scaling_channels(module1, module2, module1_output_channel_axis, module2_input_channel_axis)
+
+def equalize(model, paired_modules_list, threshold):
+    ''' Given a list of adjacent modules within a model, equalization will
+    be applied between each pair, this will repeated until convergence is achieved
+
+    Keeps a copy of the changing modules from the previous iteration, if the copies
+    are not that different than the current modules (determined by converged_test),
+    then the modules have converged enough that further equalizing is not necessary
+
+    '''
+    name_to_module = {}
+    copies = {}
+    name_set = {name for pair in paired_modules_list for name in pair}
+
+    for name, module in model.named_modules():
+        if name in name_set:
+            name_to_module[name] = module
+            copies[name] = None
```

Comment:
copies sounds vague, would be good to rename it to something like `previous_name_to_module`